File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int n = nums.size();
        vector<int> res(n, 0);
        int even = 0;
        int odd = 1;
        for(int i = 0; i < n; i++) {
            if(nums[i]>=0) res[even] = nums[i], even+=2;
            else res[odd] = nums[i], odd+=2;
        }
        return res;
    }
};

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var rearrangeArray = function(nums) {
    let res = Array(nums.length).fill(0);
    let pos = 0;
    let neg = 1;
    for(let i = 0; i < nums.length; i++) {
        if(nums[i] < 0) {
            res[neg] = nums[i];
            neg+=2;
        }
        else {
            res[pos] = nums[i];
            pos+=2;
        }
    }
    return res;
};